tests/integration/helpers_deployments.py

```diff
@@ -321,37 +322,35 @@ async def wait_until(  # noqa: C901
             if app not in wait_for_exact_units:
                 wait_for_exact_units[app] = 1
 
-    try:
-        logger.info("\n\n\n")
-        logger.info(
-            subprocess.check_output(
-                f"juju status --model {ops_test.model.info.name}", shell=True
-            ).decode("utf-8")
+    logger.info("\n\n\n")
+    logger.info(
+        subprocess.check_output(
+            f"juju status --model {ops_test.model.info.name}", shell=True
+        ).decode("utf-8")
+    )
+
+    start = datetime.now()
+    while start + timedelta(seconds=timeout) > datetime.now():
+        sleep(10)
```

Comment:
I have no explanation for this.

With `Retry()` this loop breaks. I've spent AGES on this thing.
Works on `main`, breaks here. No exception. Same tenacity version.
NOT related to the change on line 26.
I don't understand. After one loop the execution simply stops with `Retry()` 
Works perfectly with the "simulation" of the tenacity functionality.
If any mis-behavior on charm side: this dead simple execution flow surely should let it escalate. (So far nothing observed, I got green pipelines again.)